### ISSUE TITLE
EKS 1.15 / amazon-vpc-cni-k8s v1.7.5: "Readiness probe failed"

### ISSUE THREAD

**Issue by safaa-alnabulsi** [aws/amazon-vpc-cni-k8s#1267]
**What happened**:

Hello everyone, 
In the process of upgrading my K8s cluster from v1.15 to v1.16. I wanted to make sure all my controllers are on the recommended version by aws in [this page](https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html). I upgraded kube-proxy and coredns to the recommended versions: 

kube-proxy: v1.6.12 (works)
core-dns: 1.6.6 (works)
amazon-vpc-cni-k8s: 1.7.5 (doesn't work)

The deployment is done exactly as in the release docs: https://github.com/aws/amazon-vpc-cni-k8s/releases/tag/v1.7.5
I tried all the solutions mentioned in this repo:
1. downgrading kube-proxy from 1.6.13 to 1.6.12 with/without adding this label `readOnlyRootFilesystem: false `. 
2. I tried the following versions of amazon-vpc-cni-k8s: v1.7.5, v1.7.4, v1.7.3, v1.7.2 and finally tried to roll-back to 1.6.3 but the issue is always the same.

```
  Type     Reason     Age                   From                                                   Message
  ----     ------     ----                  ----                                                   -------
  Normal   Scheduled  3m56s                 default-scheduler                                      Successfully assigned kube-system/aws-node-98fnz to ip-some-ip..eu-west-1.compute.internal
  Normal   Pulling    3m54s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Pulling image "602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni-init:v1.7.5"
  Normal   Pulled     3m39s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Successfully pulled image "602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni-init:v1.7.5"
  Normal   Created    3m38s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Created container aws-vpc-cni-init
  Normal   Started    3m38s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Started container aws-vpc-cni-init
  Normal   Started    3m29s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Started container aws-node
  Warning  Unhealthy  3m25s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Readiness probe failed: {"level":"info","ts":"2020-10-20T13:57:04.975Z","caller":"/usr/local/go/src/runtime/proc.go:203","msg":"timeout: failed to connect service \":50051\" within 1s"}
  Warning  Unhealthy  3m15s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Readiness probe failed: {"level":"info","ts":"2020-10-20T13:57:14.998Z","caller":"/usr/local/go/src/runtime/proc.go:203","msg":"timeout: failed to connect service \":50051\" within 1s"}
  Warning  Unhealthy  3m6s                  kubelet, ip-some-ip.eu-west-1.compute.internal  Readiness probe failed: {"level":"info","ts":"2020-10-20T13:57:24.966Z","caller":"/usr/local/go/src/runtime/proc.go:203","msg":"timeout: failed to connect service \":50051\" within 1s"}
  Warning  Unhealthy  2m55s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Readiness probe failed: {"level":"info","ts":"2020-10-20T13:57:34.954Z","caller":"/usr/local/go/src/runtime/proc.go:203","msg":"timeout: failed to connect service \":50051\" within 1s"}
  Warning  Unhealthy  2m45s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Readiness probe failed: {"level":"info","ts":"2020-10-20T13:57:45.006Z","caller":"/usr/local/go/src/runtime/proc.go:203","msg":"timeout: failed to connect service \":50051\" within 1s"}
  Warning  Unhealthy  2m36s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Readiness probe failed: {"level":"info","ts":"2020-10-20T13:57:54.957Z","caller":"/usr/local/go/src/runtime/proc.go:203","msg":"timeout: failed to connect service \":50051\" within 1s"}
  Warning  Unhealthy  2m25s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Readiness probe failed: {"level":"info","ts":"2020-10-20T13:58:04.959Z","caller":"/usr/local/go/src/runtime/proc.go:203","msg":"timeout: failed to connect service \":50051\" within 1s"}
  Warning  Unhealthy  2m25s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Liveness probe failed: {"level":"info","ts":"2020-10-20T13:58:05.694Z","caller":"/usr/local/go/src/runtime/proc.go:203","msg":"timeout: failed to connect service \":50051\" within 1s"}
  Warning  Unhealthy  2m16s                 kubelet, ip-some-ip..eu-west-1.compute.internal  Readiness probe failed: {"level":"info","ts":"2020-10-20T13:58:14.960Z","caller":"/usr/local/go/src/runtime/proc.go:203","msg":"timeout: failed to connect service \":50051\" within 1s"}
  Normal   Killing    2m5s                  kubelet, ip-some-ip..eu-west-1.compute.internal  Container aws-node failed liveness probe, will be restarted
  Warning  Unhealthy  116s (x4 over 2m15s)  kubelet, ip-some-ip..eu-west-1.compute.internal  (combined from similar events): Readiness probe failed: {"level":"info","ts":"2020-10-20T13:58:34.963Z","caller":"/usr/local/go/src/runtime/proc.go:203","msg":"timeout: failed to connect service \":50051\" within 1s"}
  Normal   Pulling    115s (x2 over 3m37s)  kubelet, ip-some-ip.eu-west-1.compute.internal  Pulling image "602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:v1.7.5"
  Normal   Created    113s (x2 over 3m29s)  kubelet, ip-some-ip..eu-west-1.compute.internal  Created container aws-node
  Normal   Pulled     113s (x2 over 3m30s)  kubelet, ip-some-ip..eu-west-1.compute.internal  Successfully pulled image "602401143452.dkr.ecr.us-west-2.amazonaws.com/amazon-k8s-cni:v1.7.5"
```
The containers keep restarting after they reach "CrashLoopBackOff" state: 
```
 kubectl get pod  | grep aws-node
aws-node-7skbm                                               1/1     Running            0          5h48m
aws-node-fslj7                                               1/1     Running            0          6h15m
aws-node-lz89h                                               0/1     Running            3          5m55s
aws-node-mblnp                                               0/1     Running            7          16m
aws-node-nx45r                                               0/1     CrashLoopBackOff   9          26m
aws-node-wc5fb                                               1/1     Running            0          5h34m
``` 
**What you expected to happen**:
I expected the recommended version by aws to work together with the rest of controllers. Espically that version v1.7.5 of amazon-vpc-cni-k8s is recommended for all k8s versions.

**How to reproduce it (as minimally and precisely as possible)**:
It seems the version of amazon-vpc-cni-k8s is not compatible with kube-proxy. 
I used exactly the versions provided in the docs without any change.

- Kube-proxy: following this doc: https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html#1-16-prerequisites downloaded from here: https://amazon-eks.s3-us-west-2.amazonaws.com/cloudformation/2020-06-10/kube-proxy-daemonset.yaml 

- amazon-vpc-cni:  following https://github.com/aws/amazon-vpc-cni-k8s/releases/tag/v1.7.5
downloaded from here https://raw.githubusercontent.com/aws/amazon-vpc-cni-k8s/v1.7.5/config/v1.7/aws-k8s-cni.yaml

and simply kubectl apply -f $file ... 

- eks: v1.15
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):  
```
$ kubectl version --short
Client Version: v1.15.0
Server Version: v1.15.11-eks-065dce
```
- CNI Version: v1.7.5
- coredns: 1.6.6
- OS (e.g: `cat /etc/os-release`):
```
cat /etc/os-release
NAME="Amazon Linux"
VERSION="2"
ID="amzn"
ID_LIKE="centos rhel fedora"
VERSION_ID="2"
PRETTY_NAME="Amazon Linux 2"
ANSI_COLOR="0;33"
CPE_NAME="cpe:2.3:o:amazon:amazon_linux:2"
HOME_URL="https://amazonlinux.com/"

```

- Kernel (e.g. `uname -a`):
```

uname -a
Linux ip-someip.eu-west-1.compute.internal 4.14.198-152.320.amzn2.x86_64 #1 SMP Wed Sep 23 23:57:28 UTC 2020 x86_64 x86_64 x86_64 GNU/Linu
```x

**Comment by safaa-alnabulsi** [comment]
I'm closing this issue as i found it's not related to vpc-cni but to kueb-proxy. 
After checking the logs and this [doc page](https://docs.aws.amazon.com/eks/latest/userguide/update-cluster.html#1-16-prerequisites
), I noticed that I missed replacing this value MASTER_ENDPOINT besides the region. 
`
aws eks describe-cluster     --name s24-infinity-cluster-dev     --region eu-west-1     --query 'cluster.endpoint'     --output text`

However, that's not possible as we have many clusters. So I removed the other configs from the deployment file of kube-proxy and left only the DaemonSet.
That solved my problem. All pods are restarting normally now.

Thank you.
